backend/src/akvo/lumen/lib/transformation/delete_column.clj

```diff
@@ -21,17 +24,33 @@
        (filter (fn [[distinct-columns _]]
                  (not-empty (filter #(= % column-name) distinct-columns))))))
 
+(defn visualisations-with-dataset-column [tenant-conn dataset-id column-name]
+ (->> (visualisation/visualisations-by-dataset-id tenant-conn dataset-id)
+      (map #(let [{:keys [spec visualisationType id name]} %
+                  columns (aggregation.commons/spec-columns visualisationType
+                                                            (walk/keywordize-keys spec)
+                                                            dataset-id)]
+              [id name columns]))
+      (filter (fn [[id name columns]]
+                (some #(= % column-name) columns)))))
+
 (defmethod engine/apply-operation "core/delete-column"
   [{:keys [tenant-conn]} table-name columns op-spec]
   (let [column-name (col-name op-spec)
         merged-sources (merged-sources-with-column tenant-conn column-name (:dataset-id op-spec))]
     (if (empty? merged-sources)
-      (let [column-idx  (engine/column-index columns column-name)]
-        (db.tx.engine/delete-column tenant-conn {:table-name table-name :column-name column-name})
-        {:success?      true
-         :execution-log [(format "Deleted column %s" column-name)]
-         :columns       (into (vec (take column-idx columns))
-                              (drop (inc column-idx) columns))})
+      (if-let [existent-viss (seq (visualisations-with-dataset-column tenant-conn (:dataset-id op-spec) column-name))]        
+        {:success? false
+         :message  (format "Cannot delete column. It is used in the following visalisations: %s"
```

Comment:
This user message needs to be _localized_. How are we going to do that?

Reply:
https://github.com/akvo/akvo-lumen/issues/2646